Show realm name on cards grid card

FadhlanR opened the pull request:
https://github.com/cardstack/boxel/assets/12637010/6488010e-8fce-48cc-9759-4afddad649f3

(EDIT)


I decided to use a different approach to display the title of the cards grid. Unlike other card titles that are defined in the operator mode header, the title of the cards grid is defined in the card itself from the realm name field.

FadhlanR committed:
Show realm name on cards grid

jurgenwerk reviewed (commented):
I'm trying to run this branch locally but the realm name does not show up. What am I missing?

jurgenwerk commented on packages/host/app/styles/app.css:
```diff
@@ -386,9 +386,17 @@ nav > .directory-level {
+.cards-grid__title {
+  color: var(--boxel-header-text-color, var(--boxel-dark));
+  font: 700 var(--boxel-font-lg);
+  letter-spacing: normal;
```
Is this needed?

FadhlanR replied:
removed

jurgenwerk commented on packages/host/app/styles/app.css:
```diff
@@ -386,9 +386,17 @@ nav > .directory-level {
+  color: var(--boxel-header-text-color, var(--boxel-dark));
+  font: 700 var(--boxel-font-lg);
+  letter-spacing: normal;
+  text-transform: none;
```
Is this needed?

FadhlanR replied:
removed

jurgenwerk commented on packages/host/app/styles/app.css:
```diff
@@ -398,14 +406,27 @@ nav > .directory-level {
+  width: 8px;
+}
+.cards-grid__cards::-webkit-scrollbar-track {
+  background-color: transparent; /* Make the scrollbar track transparent */
```
These CSS comments are not necessary since they almost exactly repeat what is already described in css definitions

jurgenwerk replied:
Do we really need to manipulate the looks of the scrollbar?

FadhlanR replied:
You're right, we don't need it for now.

FadhlanR replied:
removed

jurgenwerk commented on packages/host/app/styles/app.css:
```diff
@@ -398,14 +406,27 @@ nav > .directory-level {
+
+  overflow: auto;
+}
+.cards-grid__cards::-webkit-scrollbar {
```
Why do these apply to webkit only?

FadhlanR replied:
removed

jurgenwerk commented on packages/host/app/styles/app.css:
```diff
@@ -513,11 +548,26 @@ nav > .directory-level {
   pointer-events: auto;
 }
 
+.operator-mode-card-stack__card--cards-grid {
```
`--` notation is used for describing a state or a version of the element. `cards-grid` is a block so this brings some confusion

FadhlanR replied:
In this case, cards-grid is a state that differentiates the styles of the cards grid and other cards in operator mode.

jurgenwerk commented on packages/host/app/templates/card.hbs:
```diff
@@ -1,4 +1,6 @@
-<this.getIsolatedComponent />
+<div class="card-isolated-component">
```
Unrelated to this PR (and not requesting changes on this), but I really wish we had a utility css class approach. This way we could just use something like:

```
<div class="sp-lg">
  <this.getIsolatedComponent />
</div>
```

With this we could avoid inventing a new class almost every time we needed to add some padding to an element or similar basic stuff.

burieberry replied:
Let's have another discussion about this during a boxel mob session or retro. Maybe we can just have some utility classes and restrict it for host app use. Utility classes idea may not work with css-encapsulation changes that are coming, though.

jurgenwerk reviewed (changes requested):
1. There is design inconsistency:

<img width="210" alt="image" src="https://github.com/cardstack/boxel/assets/273660/3428d97f-0d02-44ab-8926-847313d4d91e">

2. It would be great to have a test for this

FadhlanR committed:
Update styles

FadhlanR committed:
Fix inconsistency

FadhlanR committed:
Add test

FadhlanR commented:
@jurgenwerk you have to ensure the latest runtime common is installed.

Fixed inconsistency design. https://github.com/cardstack/boxel/pull/393/commits/6ce506c716b47303a28266632b24cc9a7eb10269
Added a test. https://github.com/cardstack/boxel/pull/393/commits/93843fe4f8464d0132890c52ddc02f4a609d7359

FadhlanR committed:
Merge branch 'main' into cs-5640-show-realm-name-on-cards-grid

FadhlanR committed:
Handle cards grid edit

FadhlanR committed:
revert wrong update

burieberry reviewed (commented):
I think it would be nice if we could use the name of the realm as the CardsGrid card's display name, instead of calling it `Cards Grid`. Conceptually, it tracks because that card's sole purpose is to display the realm's cards. However, implementation-wise, I realize that `displayName` is a static property.

FadhlanR commented:
@burieberry I think we could add a new static function in CardBase class and override it in CardsGrid class. Here is an example.

CardBase
```
static getDisplayName(
    instance: CardBase,
  ): any {
    return instance.constructor.displayName;
  }
```

CardsGrid
```
static getDisplayName(
    instance: CardBase,
  ): any {
    return instance[realmInfo]?.name;
  }
```

habdelra reviewed (approved):
looks good to me (pending Matic's review comments)

FadhlanR committed:
Merge branch 'main' into cs-5640-show-realm-name-on-cards-grid

FadhlanR committed:
merge resolve

FadhlanR committed:
revert unrelated changes
